Gordon360/Controllers/RecIMControllers/LeagueController.cs

```diff
@@ -0,0 +1,114 @@
+﻿using Gordon360.Models.CCT;
+using Gordon360.Services;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+
+namespace Gordon360.Controllers.RecIMControllers
+{
+    [Route("api/recim")]
+    public class LeagueController : GordonControllerBase
+    {
+        private readonly ILeagueService _leagueService;
+
+        public LeagueController(ILeagueService leagueService)
+        {
+            _leagueService = leagueService;
+        }
+
+        ///<summary>Gets a list of all Leagues</summary>
+        /// <returns>
+        /// All Existing Leagues 
+        /// </returns>
+        [HttpGet]
+        [Route("league")]
+        public ActionResult<IEnumerable<League>> GetAllLeagues()
+        {
+            var result = _leagueService.GetAllLeagues();
+
+            if (result == null)
+            {
+                return NotFound();
+            }
+            return Ok(result);
+        }
+
+        ///<summary>Gets a League object by ID number</summary>
+        /// <param name="leagueID">League ID Number</param>
+        /// <returns>
+        /// League object
+        /// </returns>
+        [HttpGet]
+        [Route("league/{leagueID}")]
+        public ActionResult<League> GetLeagueByID(int leagueID)
+        {
+            var result = _leagueService.GetLeagueByID(leagueID);
+
+            if (result == null)
+            {
+                return NotFound();
+            }
+            return Ok(result);
+        }
+
+        ///<summary>Gets a list of all Series associated with a League</summary>
+        /// <param name="leagueID"> League ID Number</param>
+        /// <returns>
+        /// list of all Series object associated with a league
+        /// </returns>
+        [HttpGet]
+        [Route("league/{ID}/series")]
+        public ActionResult<HashSet<Series>> GetAllLeagueSeries(int leagueID)
```

Comment:
You don't need to specify this as explicitly a `HashSet`, that constrains the type of the collection unnecessarily, and will likely make things difficult for you in the future. The actual type of `League.Series` is `ICollection<Series>`, meaning some collection that implements the generic `ICollection` interface. I suspect that the reason you're using `HashSet` here is because that's what is used in the default constructor generated by EF Core, but that is likely not the exact type of collection that EF Core will use for this data at runtime.

 ```suggestion
        public ActionResult<ICollection<Series>> GetAllLeagueSeries(int leagueID)
```

Reply:
Switched the HashSets to Lists as those are the easiest collections to return from EF Core. However, I don't know if it is better yet to return an ICollection

Reply:
`List` implements `ICollection`, so you can use `ICollection` and still return a list. However, if you specify the type as `List`, you won't be able to use any other collection type.

Reply:
Ohhh, I see what you mean now. I thought they were statically typed. I just made that adjustment to all List returns.